arch/arm/boot/dts/imx6ul-ts7180.dts

```diff
@@ -0,0 +1,1020 @@
+			compatible = "regulator-fixed";
+			regulator-name = "en-can";
+			regulator-min-microvolt = <3000000>;
+			regulator-max-microvolt = <3000000>;
+			gpio = <&gpio6 20 GPIO_ACTIVE_HIGH>;
+		};
+
+	reg_3v3: regulator-3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		pinctrl-0 = <>; /* still needed if always-on? */
+	};
+
+        i2c_gpio@0 {
+	        compatible = "i2c-gpio";
+	        clock-frequency = <100000>;
+	        pinctrl-names = "default";
+	        pinctrl-0 = <&pinctrl_i2c_gpio>;
+		gpios = <
+			&gpio3 9 GPIO_ACTIVE_HIGH /* I2C_DAT on HD12 pin 1 */
+			&gpio3 8 GPIO_ACTIVE_HIGH /* I2C_CLK on HD12 pin 6 */
+			>;
+	        status = "okay";
+		i2c-gpio,delay-us = <2>;	/* ~100 kHz */
+	        #address-cells = <1>;
+	        #size-cells = <0>;
+        };
+
+	pps {
+		compatible = "pps-gpio";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ppsgpio>;
+		gpios = <&gpio5 1 GPIO_ACTIVE_HIGH>;
+
+		status = "okay";
+	};
+};
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio1 2 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio1 3 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	status = "okay";
+
+	/*
+	 * lis2mdl_magn (now lis2mdl) is handled by the st_magn driver
+	 *
+	 *	- MAG_N_IRQ into NAND_ALE/SD2_RST#/PWM3 pin B4
+	 *	- as well as the FPGA
+	 *
+	 * drdy-int-pin is mandatory, not optional, when using interrupts.
+	 */
+	magn: magnetometer@1e {
+		compatible = "st,lis2mdl";
+		reg = <0x1e>;
+		st,drdy-int-pin = <1>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <10 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_NAND_ALE__GPIO4_IO10 */
+		vdd-supply = <&reg_3v3>;
+		vddio-supply = <&reg_3v3>;
+		status = "okay";
+	};
+
+	/*
+	 * ism330dlc is (now) handled by the lsm6dsx gyro/accel driver
+	 *
+	 *	- GYRO_INT into CSI_HSYNC pin F3
+	 */
+	gyro: ism330@6a {
+		compatible = "st,ism330dlc";
+		reg = <0x6a>;
+		interrupt-parent = <&gpio4>;
+		interrupts = <20 IRQ_TYPE_LEVEL_HIGH>;    /* See MX6UL_PAD_CSI_HSYNC__GPIO4_IO20 */
+		vdd-supply = <&reg_3v3>;
+		status = "okay";
+	};
+
+	rtc: m41t00s@68 {
+		compatible = "m41t00";
+		reg = <0x68>;
+	};
+
+	supervisor: supervisor@54 {
+		compatible = "technologic,ts7100-wdt";
+		/* 5min timeout default, in case of slow userspace.
+		 * Set to 0 to disable WDT at startup. Userspace procs
+		 * can still start feeding with different values later -
+                 * watchdog(8) defaults to 60 seconds.
+		 */
+		timeout-sec = <300>;
+		reg = <0x54>;
+                status = "okay";
+	};
+};
+
+&adc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_adc>;
+	vref-supply = <&reg_adc_vref>;
+	num-channels = <10>;
+	status = "okay";
+};
+
+&cpu0 {
+	arm-supply = <&reg_arm>;
+	soc-supply = <&reg_soc>;
+	dc-supply = <&reg_gpio_dvfs>;
+};
```

Comment:
Check these, I do not think upstream kernel uses these. I think this was only in NXP specific kernels trees.